helix-term/src/ui/picker.rs

```diff
@@ -762,13 +776,96 @@ impl<T: Item + 'static> Picker<T> {
             );
         }
     }
+
+    fn render_title(
+        &mut self,
+        mode: PickerTitle,
+        area: Rect,
+        surface: &mut Surface,
+        cx: &mut Context,
+    ) {
+        let borders = BorderType::line_symbols(BorderType::Plain);
+        let background = cx.editor.theme.get("ui.background");
+        let text_style = cx.editor.theme.get("ui.text").add_modifier(Modifier::BOLD);
+        // Add four for margin and bounding box edges
+        let title_width = self.title.len() as u16 + 4;
+        match mode {
+            PickerTitle::Center => {
+                const TITLE_BOX_HEIGHT: u16 = 2;
+
+                if area.y < TITLE_BOX_HEIGHT || area.width <= title_width + 1 {
+                    return;
+                }
+
+                // Compute area for title rendering
+                let area_to_clip = area.width - title_width;
+                let mut area = area
+                    .clip_left(area_to_clip / 2)
+                    .clip_right(area_to_clip / 2)
+                    .with_height(TITLE_BOX_HEIGHT);
+                area.y -= TITLE_BOX_HEIGHT;
+
+                // -- Render the frame:
+                // clear area
+                surface.clear_with(area, background);
+
+                // render the border
+                let block = Block::default().borders(Borders::TOP | Borders::LEFT | Borders::RIGHT);
+                block.render(area, surface);
+
+                // render the title text
+                surface.set_string(
+                    // Add two for spacing for border box and margin
+                    area.x + 2,
+                    area.y + 1,
+                    &self.title,
+                    text_style,
+                );
```

Comment:
Yes I think the picker title will overflow if it is too long, I remember hitting this with inline diagnostics